cypress-tests/cypress/support/commands/projects.ts

```diff
@@ -69,39 +69,46 @@ interface NewProjectProps extends ProjectIdentifier {
 }
 
 function createProject(newProjectProps: NewProjectProps) {
-  cy.visit("/projects/new");
-  cy.getDataCy("field-group-title")
-    .should("be.visible")
-    .clear()
-    .type(newProjectProps.name);
-  if (newProjectProps.namespace) {
-    cy.get("#namespace-input")
-      .should("be.visible")
-      .clear()
-      .type(newProjectProps.namespace);
-  }
-
-  if (newProjectProps.templateName)
-    cy.contains(newProjectProps.templateName).should("be.visible").click();
-
-  if (newProjectProps.visibility) {
-    cy.getDataCy(`visibility-${newProjectProps.visibility}`)
-      .should("be.visible")
-      .click();
-  }
-
-  // The button may take some time before it is clickable
-  cy.get("[data-cy=create-project-button]", { timeout: TIMEOUTS.vlong })
-    .should("be.enabled")
-    .click();
-  cy.url({ timeout: TIMEOUTS.vlong }).should(
-    "contain",
-    newProjectProps.name.toLowerCase()
-  );
-  cy.get(`[data-cy="header-project"]`, { timeout: TIMEOUTS.vlong }).should(
-    "be.visible"
-  );
-  cy.get("ul.nav-pills-underline").should("be.visible");
+  const namespace = newProjectProps.namespace ?? Cypress.env("TEST_USERNAME");
+  const slug = encodeURIComponent(`${namespace}/${newProjectProps.name}`);
+  cy.request({failOnStatusCode: false, method: "GET", url: `/ui-server/api/projects/${slug}`}).then((response) => {
```

Comment:
I feel this change is unnecessary since we can re-run single file specs but not single blocks: no need to recheck the project every time.
Do you mind renaming the function to something more descriptive like `createProjectWhenMissing` or `checkAndCreateProject`? I added a couple of comments to the spec files review before realizing what it actually does :grin:

Reply:
@lorenzo-cavazzi this is the entire point of this PR. It's to move the project creation into `beforeEach` blocks so that the project creation hook can be re-attempted if it fails.

Reply:
Exactly what @leafty said.